Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -2480,8 +2488,8 @@ - (void)_updateVisibleContentRects
 
     CGFloat scaleFactor = contentZoomScale(self);
     CGRect unobscuredRect = UIEdgeInsetsInsetRect(self.bounds, computedContentInsetUnadjustedForKeyboard);
-    CGRect unobscuredRectInContentCoordinates = _perProcessState.frozenUnobscuredContentRect ? _perProcessState.frozenUnobscuredContentRect.value() : [self convertRect:unobscuredRect toView:_contentView.get()];
-    unobscuredRectInContentCoordinates = CGRectIntersection(unobscuredRectInContentCoordinates, [self _contentBoundsExtendedForRubberbandingWithScale:scaleFactor]);
+    WebCore::FloatRect unobscuredRectInContentCoordinates = WebCore::FloatRect(_perProcessState.frozenUnobscuredContentRect ? _perProcessState.frozenUnobscuredContentRect.value() : [self convertRect:unobscuredRect toView:_contentView.get()]);
+    unobscuredRectInContentCoordinates.intersect([self _contentBoundsExtendedForRubberbandingWithScale:scaleFactor]);
```

Comment:
Is this correct? The PR description says:

```
This PR fixes this by checking if the CGRect is `CGRectNull`, and if so,
replaces its origin with the original unobscured rect's origin.
```

Does `intersect` preserve the original origin? If not, wouldn't this cause an unexpected scroll to (0, 0)?

Reply:
Yes, the comment was before I applied Tim's suggestion. It does preserve the original origin; I tested it manually.